src/Statement.php

```diff
@@ -237,28 +303,19 @@ public function verify($options = array()) {
 
         $header = $jws->getHeader();
 
-        //
-        // there is a JWS spec security issue with allowing non-RS algorithms
-        // to be specified and it is against the Tin Can spec anyways so we
-        // want to fail hard on non-RS algorithms
-        //
-        if (! in_array($header['alg'], array('RS256', 'RS384', 'RS512'), true)) {
-            throw new \InvalidArgumentException("Refusing to verify signature: Invalid signing algorithm ('" . $options['algorithm'] . "')");
-        }
+        self::validateNameOfRSAlgorithm($header['alg']);
 
         if (isset($options['publicKey'])) {
             $publicKeyFile = $options['publicKey'];
         }
-        else if (isset($header['x5c'])) {
+        elseif (isset($header['x5c'])) {
             $cert = "-----BEGIN CERTIFICATE-----\r\n" . chunk_split($header['x5c'][0], 64, "\r\n") . "-----END CERTIFICATE-----\r\n";
             $cert = openssl_x509_read($cert);
             if (! $cert) {
                 return array('success' => false, 'reason' => 'failed to read cert in x5c: ' . openssl_error_string());
             }
+            
```

Comment:
I've removed the check here because it seems like if $cert is valid then this line will always execute successfully.